libraries/apollo-debug-server/src/androidMain/kotlin/com/apollographql/apollo3/debugserver/internal/server/Server.android.kt

```diff
@@ -0,0 +1,105 @@
+package com.apollographql.apollo3.debugserver.internal.server
+
+import android.net.LocalServerSocket
+import android.net.LocalSocket
+import com.apollographql.apollo3.ApolloClient
+import com.apollographql.apollo3.debugserver.internal.graphql.GraphQL
+import com.apollographql.apollo3.debugserver.internal.initializer.ApolloDebugServerInitializer
+import kotlinx.coroutines.CancellationException
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.SupervisorJob
+import kotlinx.coroutines.asCoroutineDispatcher
+import kotlinx.coroutines.cancel
+import kotlinx.coroutines.launch
+import java.io.BufferedReader
+import java.io.PrintStream
+import java.util.concurrent.Executors
+
+internal actual fun createServer(
+    apolloClients: Map<ApolloClient, String>,
+): Server = AndroidServer(apolloClients)
+
+private class AndroidServer(
+    apolloClients: Map<ApolloClient, String>,
+) : Server {
+  companion object {
+    private const val SOCKET_NAME_PREFIX = "apollo_debug_"
+  }
+
+  private var localServerSocket: LocalServerSocket? = null
+  private val dispatcher = Executors.newCachedThreadPool().asCoroutineDispatcher()
+  private val coroutineScope = CoroutineScope(SupervisorJob() + dispatcher)
+
+  private val graphQL = GraphQL(apolloClients)
+
+  override fun start() {
+    if (localServerSocket != null) error("Already started")
+    val localServerSocket = LocalServerSocket("$SOCKET_NAME_PREFIX${ApolloDebugServerInitializer.packageName}")
+    this.localServerSocket = localServerSocket
+    coroutineScope.launch {
+      while (true) {
+        val clientSocket = try {
+          localServerSocket.accept()
+        } catch (_: Exception) {
+          // Server socket has been closed (stop() was called)
+          break
+        }
+        launch { handleClient(clientSocket) }
+      }
+    }
```

Comment:
Does this need to be a coroutine? Is there any other suspending call than `ApolloStore.dump()`? This one really shouldn't be supend as it's synchronous. It's part of the APIs I want to change for v4. So all in all, unless there are other, I would just wrap the dump in `runBlocking {apolloStore.dump()}`

Reply:
Well, I just got reminded by my IDE that `runBlocking` doesn't exist in `commonMain` 😅

Reply:
IIRC if you remove the JS target, it will "work" again. By the time we add JS again, we should remove the suspend from `ApolloStore.dump`

Reply:
>if you remove the JS target, it will "work" again

wow, indeed 🤯

Reply:
That being said (⚠️ can of worms), would it be cool for `Resolver.resolve()` to be suspend, and allow `@ApolloObject` funs to be suspend?

Reply:
"probably" but this goes into data loader land and parallel execution questions. 
If a field is suspend, is it sharing its thread with other fields? If not I think it gives the wrong impression because you could as well call `runBlocking {}` inside the field logic (except for JS...). TLDR; might be useful but I don't really want to open that door at that point.